.github/workflows/test-upstream.yaml

```diff
@@ -113,22 +114,18 @@ jobs:
           git fetch upstream --tags
 
       - name: set up python
-        uses: actions/setup-python@v5
+        uses: actions/setup-python@v5.5
         with:
           python-version: ${{ matrix.python-version }}
 
       - name: Install poetry
-        run: pipx install poetry==1.*
-
-      - name: Check Poetry version
-        run: poetry --version
-
-      - name: Install dynamic versioning plugin
-        run: poetry self add "poetry-dynamic-versioning[plugin]"
+        run: |
+          pipx install poetry
+          pipx inject poetry poetry-dynamic-versioning
 
       - name: Load cached venv
         id: cached-poetry-dependencies
-        uses: actions/cache@v3
+        uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57
```

Comment:
Why do some actions use specific commits but others use versions? Can we standardise?

Reply:
Done.